### ISSUE TITLE
Extraneous non-props attributes (errors) were passed to component 

### ISSUE THREAD

**Issue by Livijn** [inertiajs/inertia#543]
### Versions:

- `@inertiajs/inertia` version: 0.8.5
- `@inertiajs/inertia-vue3` version: 0.3.5

### Describe the problem:
I am receiving this warning when just using this `return Inertia::render('About')`
> Extraneous non-props attributes (errors) were passed to component but could not be automatically inherited because component renders fragment or text root nodes.

**Comment by Livijn** [comment]
Ok, makes sense. Where is the `error` prop coming from though? I wouldn't mind omitting it, since I don't use it.

**Comment by reinink** [comment]
@Livijn It's a default prop set by the Laravel adapter, used for passing along validation errors (see [here](https://github.com/inertiajs/inertia-laravel/blob/master/src/Middleware.php#L44-L51)).

You can remove it by simply omitting the default shared props in your `HandleInertiaRequests` middleware (see [here](https://github.com/inertiajs/pingcrm/blob/master/app/Http/Middleware/HandleInertiaRequests.php#L39)).

**Comment by reinink** [comment]
Hmm, so I'm realizing another way to remove these warnings is to simply [disabling attribute inheritance](https://v3.vuejs.org/guide/component-attrs.html#disabling-attribute-inheritance) on the page component. You can do this by setting `inheritAttrs` to `false`, like this:

```js
<script>
export default {
  inheritAttrs: false,
  // ...
}
</script>
```

It makes me wonder if there is a way for us to do this automatically at the Vue level. 🤔

**Comment by Livijn** [comment]
I decided to just delete all the shared props in the `transformProps` and send them to a store.

```ts
    transformProps: (props: IObjectKeys) => {
      store.setup({ prop: props.sharedProp });

      delete props.sharedProp;
      delete props.errors;

      return props;
    },
```

**Comment by reinink** [comment]
@Livijn Just so you know, this has been fixed properly in #681. Also, FYI, there is a VERY good chance we'll be removing `transformProps` from Inertia in the near future. It's just created more problems than it solved.
